src/Features/VisualBasic/Portable/BraceCompletion/InterpolatedStringBraceCompletionService.vb

```diff
@@ -0,0 +1,64 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.BraceCompletion
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.VisualBasic
+
+<Export(LanguageNames.VisualBasic, GetType(IBraceCompletionService)), [Shared]>
+Friend Class InterpolatedStringBraceCompletionService
+    Inherits AbstractBraceCompletionService
+
+    <ImportingConstructor>
+    <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+    Public Sub New()
+        MyBase.New()
+    End Sub
+
+    Protected Overrides ReadOnly Property OpeningBrace As Char
+        Get
+            Return DoubleQuote.OpenCharacter
+        End Get
+    End Property
+
+    Protected Overrides ReadOnly Property ClosingBrace As Char
+        Get
+            Return DoubleQuote.CloseCharacter
+        End Get
+    End Property
+
+    Protected Overrides Function IsValidOpenBraceTokenAtPositionAsync(token As SyntaxToken, position As Integer, document As Document, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return Task.FromResult(IsValidOpeningBraceToken(token) AndAlso
+                   token.Span.End - 1 = position)
+    End Function
+
+    Public Overrides Function AllowOverTypeAsync(context As BraceCompletionContext, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return CheckClosingTokenKindAsync(context.Document, context.ClosingPoint, cancellationToken)
+    End Function
+
+    Public Overrides Async Function IsValidForBraceCompletionAsync(brace As Char, openingPosition As Integer, document As Document, cancellationToken As CancellationToken) As Task(Of Boolean)
+        Return OpeningBrace = brace And Await IsPositionInInterpolatedStringContextAsync(document, openingPosition, cancellationToken).ConfigureAwait(False)
+    End Function
+
+    Protected Overrides Function IsValidOpeningBraceToken(token As SyntaxToken) As Boolean
+        Return token.IsKind(SyntaxKind.DollarSignDoubleQuoteToken)
+    End Function
+
+    Protected Overrides Function IsValidClosingBraceToken(token As SyntaxToken) As Boolean
+        Return token.IsKind(SyntaxKind.DoubleQuoteToken)
+    End Function
+
+    Public Shared Async Function IsPositionInInterpolatedStringContextAsync(document As Document, position As Integer, cancellationToken As CancellationToken) As Task(Of Boolean)
+        If position = 0 Then
+            Return False
+        End If
+
+        Dim text = Await document.GetTextAsync(cancellationToken).ConfigureAwait(False)
+
+        Return text(position - 1) = "$"c
+    End Function
```

Comment:
am i wrong, or is the Vb side of this very different from the c# side.  if so, why?

Reply:
The VB side is very different from the C# side.  I don't entirely know the reasons so I did my best to just move the code and not change the logic inthis PR

Reply:
Dug into this one for a bit, the corresponding methods used by C# are either not available for VB (IsStatementContext) or do not return the expected results (IsExpressionContext) for VB.  It doesn't seem to think it's in an expression if you do `Dim s = $`.  Going to leave this as-is as I don't have a better idea of a check here.